### PR TITLE
Only load Site Health checks for persistent cache and full page cache when not available in core

### LINKED ISSUE #540: Add `can-load.php` files to load Site Health checks for persistent cache and full page cache conditionally
Since the two modules for the Site Health checks for persistent cache and full page cache got merged into core, we need to add logic so that these two modules are only loaded going forward when the relevant core functionality is not already present. For that we should use the `can-load.php` approach which was established in #293.

Relevant WordPress core commits:
* https://core.trac.wordpress.org/changeset/53955
* https://core.trac.wordpress.org/changeset/54043

Since the class that has all the changes is only loaded within WP Admin, we may not be able to check for actual functions or methods reliably, as our plugin initialization would run too early. So potentially we will simply need to check that the WordPress version is lower than 6.1.

### PR DESCRIPTION
**Pull request by mukeshpanchal27**
## Summary

Fixes #540

## Checklist

- [ ] PR has either `[Focus]` or `Infrastructure` label.
- [ ] PR has a `[Type]` label.
- [ ] PR has a milestone or the `no milestone` label.

### CODE REVIEW

File: modules/object-cache/persistent-object-cache-health-check/can-load.php [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+ * Checks whether the given module is already merged into the WordPress core.
+ *
+ * @since n.e.x.t
+ *
+ * @global string $wp_version The WordPress version string.
+ */
+return function() {
+	global $wp_version;
+
+	return ! version_compare( $wp_version, '6.1.0', '>=' );
```
Comment (felixarntz) [inline modules/object-cache/persistent-object-cache-health-check/can-load.php]: Nit-pick, but we could simplify this to
```suggestion
	return version_compare( $wp_version, '6.1.0', '<' );
```

Makes it a bit easier to understand ("is the WordPress version lower than 6.1.0?").

File: modules/site-health/audit-full-page-cache/can-load.php [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+ * Checks whether the given module is already merged into the WordPress core.
+ *
+ * @since n.e.x.t
+ *
+ * @global string $wp_version The WordPress version string.
+ */
+return function() {
+	global $wp_version;
+
+	return ! version_compare( $wp_version, '6.1.0', '>=' );
```
Comment (felixarntz) [inline modules/site-health/audit-full-page-cache/can-load.php]: See above.
```suggestion
	return version_compare( $wp_version, '6.1.0', '<' );
```

**Review by felixarntz — APPROVED** [review]
@mukeshpanchal27 Looks great, just 2 nit-picks you may want to address.

**Review by mehulkaklotar — APPROVED** [review]

**Comment by mukeshpanchal27** [comment]
Assign it to @felixarntz. It is now ready to merge.

**Pull request merged**